docs/API.md

```diff
@@ -58,6 +58,65 @@ Regex syntax: `(...)`.
 
 Captures, also known as capturing groups, extract and store parts of the matched string for later use.
 
+### `nonCaptureGroup()`
```

Comment:
@PaulJPhilp Do you think we need explicit `nonCaptureGroups` as a feature. We already do it implicitly when needed (e.g. applying quantifiers like `zeroOrMore`) to non-atomic values. What would be the benefits of having this as an explicit feature?